### PR TITLE
[10.0][ADD] Add shopinvader_membership

### PR DESCRIPTION
**Pull request by baimont**
This addon adds services to access a customer's membership lines.

### CODE REVIEW

File: shopinvader_membership/services/membership.py [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+    def _get_selection_label(self, membership_line, field):
+        """
+        Get the translated label of the membership line selection field
+        :param membership_line: memberhsip.membership_line recordset
+        :param field: str
+        :return: str
+        """
+        if field not in membership_line._fields:
+            return ""
+        # _description_selection return a list of tuple (str, str).
```
Comment (simahawk) [inline shopinvader_membership/services/membership.py]: use converto_to_export ;)
See https://github.com/shopinvader/odoo-shopinvader/blob/12.0/shopinvader/models/res_partner.py#L131

File: shopinvader_membership/services/membership.py [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+        )
+        technical_value = membership_line[field]
+        return type_dict.get(technical_value, technical_value)
+
+    def _to_json_membership_line(self, membership_line):
+        membership_line.ensure_one()
+        parser = self._get_parser_membership_line()
+        values = membership_line.jsonify(parser)[0]
+        values.update(
+            {"state": self._get_selection_label(membership_line, "state")}
```
Comment (simahawk) [inline shopinvader_membership/services/membership.py]: I'd return `{"state": {"value": ..., "label": ...}}`

File: shopinvader_membership/tests/test_membership_service.py [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+        super(TestMembershipService, self).setUp(*args, **kwargs)
+        self.membership_line_obj = self.env["membership.membership_line"]
+        self.product_obj = self.env["product.product"]
+        self.partner = self.env.ref("shopinvader.partner_1")
+        str_today = fields.Date.today()
+        self.date_today = fields.Date.from_string(str_today)
+        self.next_month = fields.Date.to_string(
+            self.date_today + timedelta(days=30)
+        )
+        self.product = self.product_obj.create(
```
Comment (simahawk) [inline shopinvader_membership/tests/test_membership_service.py]: Please move all statements but the service init to setUpClass as they will run only once.

**Review by simahawk — CHANGES_REQUESTED** [review]

**Comment by sebastienbeau** [comment]
@PierrickBrun I think it will interest you !

**Comment by PierrickBrun** [comment]
Great ! @hparfr

**Comment by baimont** [comment]
Thanks for your review @simahawk 
I made a few changes

**Review by simahawk — APPROVED** [review]

**Review by Cedric-Pigeon — APPROVED** [review]

**Comment by rousseldenis** [comment]
@baimont Could you rebase?

**Pull request merged**